### PR TITLE
allow running encryption:fix-encrypted-version for all users

### PR DESCRIPTION
**Pull request by icewind1991**
Signed-off-by: Robin Appelman <robin@icewind.nl>

### CODE REVIEW

File: apps/encryption/lib/Command/FixEncryptedVersion.php [hunk 1]
```diff
@@ -108,22 +115,42 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
+				return 1;
+			}
+
+			if ($this->userManager->get($user) === null) {
+				$output->writeln("<error>User id $user does not exist. Please provide a valid user id</error>");
+				return 1;
+			}
+
+			return $this->runForUser($user, $pathOption, $output);
+		} else if ($all) {
```
Comment (come-nc) [inline apps/encryption/lib/Command/FixEncryptedVersion.php:134]:
```suggestion
		} elseif ($all) {
```

File: apps/encryption/lib/Command/FixEncryptedVersion.php [hunk 2]
```diff
@@ -108,22 +115,42 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
+			$result = 0;
+			$this->userManager->callForSeenUsers(function(IUser $user) use ($pathOption, $output, &$result) {
+				$output->writeln("Processing files for " . $user->getUID());
+				$result = $this->runForUser($user->getUID(), $pathOption, $output);
+				if ($result !== 0) {
+					return false;
+				}
```
Comment (come-nc) [inline apps/encryption/lib/Command/FixEncryptedVersion.php:141]: What is this return false for? Does it stop processing?
Shouldn’t the function return true in other cases? Feels fishy to have return false vs no return.

Maybe:
return ($result === 0);

Comment (icewind1991) [reply apps/encryption/lib/Command/FixEncryptedVersion.php:141]: > What is this return false for? Does it stop processing?

Yes

> Shouldn’t the function return true in other cases? Feels fishy to have return false vs no return.

Not required, but it is nicer yes

**Review by come-nc — APPROVED** [review]

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by PVince81** [comment]
stable25: https://github.com/nextcloud/server/pull/35827

**Comment by PVince81** [comment]
stable24: https://github.com/nextcloud/server/pull/35865

**Comment by PVince81** [comment]
backporting further seems tricky as we changed more things so I'll leave this for now as there is a workaround: getting the list of all users and passing them to the command

**Pull request merged**